test/extensions/stats_sinks/hystrix/hystrix_test.cc

```diff
@@ -0,0 +1,360 @@
+                        uint64_t window_size) {
+    EXPECT_EQ(getStreamField(data_message, "rollingCountSemaphoreRejected"),
+              std::to_string(window_size * rejected_step));
+    EXPECT_EQ(getStreamField(data_message, "rollingCountSuccess"),
+              std::to_string(window_size * success_step));
+    EXPECT_EQ(getStreamField(data_message, "rollingCountTimeout"),
+              std::to_string(window_size * (timeout_step + timeout_retry_step)));
+    EXPECT_EQ(getStreamField(data_message, "errorCount"),
+              std::to_string(window_size *
+                             (error_step - timeout_step))); // Note that on regular operation,
+    // 5xx and timeout are raised together, so timeouts are reduced from 5xx count
+    uint64_t total = error_step + success_step + rejected_step + timeout_retry_step;
+    EXPECT_EQ(getStreamField(data_message, "requestCount"), std::to_string(window_size * total));
+    if (total != 0) {
+      EXPECT_EQ(
+          getStreamField(data_message, "errorPercentage"),
+          std::to_string(static_cast<uint64_t>(
+              100 * (static_cast<double>(total - success_step) / static_cast<double>(total)))));
+    } else {
+      EXPECT_EQ(getStreamField(data_message, "errorPercentage"), "0");
+    }
+  }
+
+  std::unordered_map<std::string, std::string> buildClusterMap(absl::string_view data_message) {
+    std::unordered_map<std::string, std::string> cluster_message_map;
+    std::vector<std::string> messages =
+        absl::StrSplit(data_message, "data:", absl::SkipWhitespace());
+    for (auto message : messages) {
+      if (absl::StrContains(getStreamField(message, "type"), "HystrixCommand")) {
+        std::string cluster_name(getStringStreamField(message, "name"));
+        cluster_message_map[cluster_name] = message;
+      }
+    }
+    return cluster_message_map;
+  }
+
+  uint64_t window_size_ = 10; // Arbitrary reasonable number.
+  const std::string cluster1_name_{"test_cluster1"};
+  ClusterTestInfo cluster1_{cluster1_name_};
+
+  NiceMock<Http::MockStreamDecoderFilterCallbacks> callbacks_;
+  NiceMock<Server::MockInstance> server_;
+  Upstream::ClusterManager::ClusterInfoMap cluster_map_;
+
+  std::unique_ptr<HystrixSink> sink_;
+  NiceMock<Stats::MockSource> source_;
+  NiceMock<Upstream::MockClusterManager> cluster_manager_;
+};
+
+TEST_F(HystrixSinkTest, EmptyFlush) {
+  InSequence s;
+  Buffer::OwnedImpl buffer = createClusterAndCallbacks();
+  // Register callback to sink.
+  sink_->registerConnection(&callbacks_);
+  sink_->flush(source_);
+  std::string data_message = TestUtility::bufferToString(buffer);
+  validate_results(data_message, 0, 0, 0, 0, 0, window_size_);
+}
+
+TEST_F(HystrixSinkTest, BasicFlow) {
+  InSequence s;
+  Buffer::OwnedImpl buffer = createClusterAndCallbacks();
+  // Register callback to sink.
+  sink_->registerConnection(&callbacks_);
+
+  // Arbitrary numbers for testing. Make sure error > timeout.
+  uint64_t success_step = 44;
```

Comment:
It is really not intuitive why these are (a) decreasing and (b) all multiples of 11.